graphbolt/src/csc_sampling_graph.cc

```diff
@@ -232,11 +265,17 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighbors(
         static_cast<int64_t>(0), std::numeric_limits<int64_t>::max());
     SamplerArgs<SamplerType::LABOR> args{indices_, random_seed, NumNodes()};
     return SampleNeighborsImpl(
-        nodes, fanouts, replace, return_eids, probs_or_mask, args);
+        nodes, return_eids,
+        GetPickFn(
+            fanouts, replace, indptr_.options(), type_per_edge_, probs_or_mask,
+            args));
   } else {
     SamplerArgs<SamplerType::NEIGHBOR> args;
     return SampleNeighborsImpl(
-        nodes, fanouts, replace, return_eids, probs_or_mask, args);
+        nodes, return_eids,
+        GetPickFn(
+            fanouts, replace, indptr_.options(), type_per_edge_, probs_or_mask,
+            args));
```

Comment:
```
PickFn pickFn;
if (layer) {
    ...
    pickFn = 
} else {
    ...
    pickFn = 
}
return SampleNeighborsImpl(nodes, return_eids, pickFn);
```

Reply:
This can't be easily done since `GetPickFn` now returns a lambda and each lambda function has its unique type. So the return values of these two `GetPickFn` have different types.